hoomd/mpcd/RotatedParallelPlateGeometry.h

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2009-2024 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+/*!
+ * \file mpcd/RotatedParallelPlateGeometry.h
+ * \brief Definition of the MPCD rotated slit channel geometry
+ */
+
+#ifndef MPCD_ROTATED_PARALLEL_PLATE_GEOMETRY_H_
+#define MPCD_ROTATED_PARALLEL_PLATE_GEOMETRY_H_
+
+#include "hoomd/BoxDim.h"
+#include "hoomd/HOOMDMath.h"
+
+#ifdef __HIPCC__
+#define HOSTDEVICE __host__ __device__ inline
+#else
+#define HOSTDEVICE inline __attribute__((always_inline))
+#include <string>
+#endif // __HIPCC__
+
+namespace hoomd
+    {
+namespace mpcd
+    {
+//! Rotated parallel plate (slit) geometry
+
+class __attribute__((visibility("default"))) RotatedParallelPlateGeometry
+    {
+    public:
+    //! Constructor
+    /*!
+     * \param separation Distance between plates
+     * \param angle Rotation angle (degrees)
+     * \param V Velocity of the wall
+     * \param no_slip Boundary condition at the wall (slip or no-slip)
+     */
+    HOSTDEVICE RotatedParallelPlateGeometry(Scalar separation, Scalar angle, Scalar V, bool no_slip)
+        : m_H(Scalar(0.5) * separation), m_angle(angle * M_PI / 180.), m_V(V), m_no_slip(no_slip)
+        {
+        }
+
+    //! Detect collision between the particle and the boundary
+    /*!
+     * \param pos Proposed particle position
+     * \param vel Proposed particle velocity
+     * \param dt Integration time remaining
+     *
+     * \returns True if a collision occurred, and false otherwise
+     *
+     * \post The particle position \a pos is moved to the point of reflection, the velocity \a vel
+     * is updated according to the appropriate bounce back rule, and the integration time \a dt is
+     * decreased to the amount of time remaining.
+     */
+    HOSTDEVICE bool detectCollision(Scalar3& pos, Scalar3& vel, Scalar& dt) const
+        {
+        const Scalar sin_angle = slow::sin(m_angle);
+        const Scalar cos_angle = slow::cos(m_angle);
+        const Scalar tan_angle = sin_angle / cos_angle;
+        const Scalar y_center = pos.x * tan_angle;
+        const Scalar y1 = y_center + m_H;
+        const Scalar y2 = y_center - m_H;
+
+        const signed char sign = (char)((pos.y > y1) - (pos.y < y2));
+
```

Comment:
I like to add comments to the code if the math is somewhat not obvious!
```suggestion
        /* check if particle is in bounds or could not have left because
         * normal speed is zero, and exit early. The normal is
         * n = [-sin(angle), cos(angle)] with no z component.
         */ 
        const signed char sign = (char)((pos.y > y1) - (pos.y < y2));
```